jupyter_core/command.py

```diff
@@ -106,17 +106,20 @@ def _execvp(cmd, argv):
 
 def _path_with_self():
     """Ensure `jupyter`'s dir is on PATH"""
-    script = sys.argv[0]
-    bindir = os.path.dirname(script)
+    scripts = [sys.argv[0]]
+    if os.path.islink(scripts[0]):
+        scripts.append(os.path.realpath(scripts[0]))
+    bindirs = [os.path.dirname(script) for script in scripts]
     path_list = (os.environ.get('PATH') or os.defpath).split(os.pathsep)
-    if (os.path.isdir(bindir)
-        and bindir not in path_list
-        and os.access(script, os.X_OK) # only if it's a script
-    ):
-        # ensure executable's dir is on PATH
-        # avoids missing subcommands when jupyter is run via absolute path
-        path_list.append(bindir)
-        os.environ['PATH'] = os.pathsep.join(path_list)
+    for bindir, script in zip(bindirs, scripts):
```

Comment:
I found this a little tricky to follow. Rather than computing bindirs and then zipping it back together with scripts, I think it would be clearer if we iterated over it once:

```python
for script in scripts:
    bindir = os.path.dirname(script)
    if (os.path.isdir(bindir)
       ...
```